datalad/metadata/tests/test_search.py

```diff
@@ -358,3 +358,265 @@ def test_meta2autofield_dict():
             'extr1': {'prop1': 'value'}}),
         {'extr1.prop1': 'value'}
     )
+
+
+def test_meta2autofield_jsonld_graph():
+    """ check proper handling of JSON-LD @graph nodes """
+    # Just a test that we would obtain the value stored for that extractor
+    # instead of what unique values it already had (whatever that means)
+    eq_(
+        _meta2autofield_dict({"r": {"@graph": ["a", "b", "c"]}}),
+        {'r.graph[0]': 'a', 'r.graph[1]': 'b', 'r.graph[2]': 'c'}
+    )
+
+
+def test_meta2autofield_jsonld_list():
+    """ check proper handling of JSON-LD @list nodes """
+    # Just a test that we would obtain the value stored for that extractor
+    # instead of what unique values it already had (whatever that means)
+    eq_(
+        _meta2autofield_dict({"r": {"@list": ["a", "b", "c"]}}),
+        {'r.list[0]': 'a', 'r.list[1]': 'b', 'r.list[2]': 'c'}
+    )
+
+
+_mocked_studyminimeta_jsonld = {
+    "@context": {
```

Comment:
Since this structure already is here, could there be a test for actual search? Just to make sure that no side effects etc?

Reply:
Good point. Adding tests for show-keys and proper search

Reply:
@yarikoptic: tests added.

Reply:
Thanks!